### PR TITLE
ci(config): add pr title linting

### PR DESCRIPTION
**Pull request by flaxel**
## Changes

- [x] add action to check pull request title using [action-semantic-pull-request](https://github.com/amannn/action-semantic-pull-request)

## Context

The titles of the pull requests are not correct in order to generate the changelog automatically. All types of changes from the [specification](https://github.com/commitizen/conventional-commit-types/blob/master/index.json) are allowed.

https://github.com/actions/stale/pull/510#issuecomment-863097970

### CODE REVIEW

**Review by C0ZEN — APPROVED** [review]
Fine with me regarding one thing: you can use a "!" to declare a breaking change.

File: .github/workflows/linting.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  title:
+  
+    runs-on: ubuntu-latest
+    
+    steps:
+      - uses: amannn/action-semantic-pull-request@v3.4.0
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          requireScope: true
```
Comment (C0ZEN) [inline .github/workflows/linting.yml:20]:
```suggestion
          types: |
            fix
            feat
            refactor
            docs
          requireScope: true
```
For this project, we would end up with those as well

Comment (C0ZEN) [reply .github/workflows/linting.yml:20]: I agree with the requirement of a scope, however, any scope.
And so, based on the documentation, I am not sure that this is possible.

**Comment by C0ZEN** [comment]
Funny because your PR title is not good based on your right ;)

**Comment by flaxel** [comment]
> Funny because your PR title is not good based on your proposition ;)

Why do you think so?

File: .github/workflows/linting.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+name: "Linting"
```
[the remark this answers is not in the document]
Comment (flaxel) [reply .github/workflows/linting.yml:1]: I used the common names to possibly just expand the action. If it needs to be more specific, I would call it `PR Check` and `lint`.

File: .github/workflows/linting.yml (hunk 1 above)
Comment (flaxel) [reply .github/workflows/linting.yml:20]: The types from the [specification](https://github.com/commitizen/conventional-commit-types/blob/master/index.json) are automatically allowed. If you want to restrict it, you can define `types`. But then I would also add `chore`.

Yes, that's unfortunately how I understood it, which is why I gave the `types`. But I think that these should be sufficient.

**Comment by C0ZEN** [comment]
The "types" refer to the white-list of types that you can use, right? In that case, you didn't include "ci".

**Comment by flaxel** [comment]
If the types are not declared, all those described in the [specification](https://github.com/commitizen/conventional-commit-types/blob/master/index.json) are taken by default.

File: .github/workflows/linting.yml (hunk 1 above)
Comment (C0ZEN) [reply .github/workflows/linting.yml:20]: Ignore my comment then

**Comment by luketomlinson** [comment]
@flaxel @C0ZEN  I am leaning against this and possibly backing out the `standard-version`  changes too. I think it introduces more friction than value, IMO.

**Comment by C0ZEN** [comment]
@luketomlinson since I am not the one making the release, honestly it's whatever you prefer.
The only important point for me is to just have a good changelog because I really think it's helpful and valuable.

**Comment by luketomlinson** [comment]
Thanks for the PR @flaxel. I think I am going to lean way from opinionated stuff like this as I think it introduces more friction than benefit in this case.

**Comment by flaxel** [comment]
No it is okay so there can be a discussion and we can find a solution together. 😁

**Pull request closed without merging**